Python 3.14 and free-threaded Python

By "support" I mean, adding them to CI. I think that will require updating the cryptography upper version pin once cryptography does a release.

When I do that locally and force pyopenssl to depend on the `main` branch of `cryptography`, I see one test failure:

```
_________________________________________________________________________________ TestConnection.test_wantWriteError _________________________________________________________________________________

self = <tests.test_ssl.TestConnection object at 0x2b676682ae0>

    def test_wantWriteError(self) -> None:
        """
        `Connection` methods which generate output raise
        `OpenSSL.SSL.WantWriteError` if writing to the connection's BIO
        fail indicating a should-write state.
        """
        client_socket, _ = socket_pair()
        # Fill up the client's send buffer so Connection won't be able to write
        # anything. Start by sending larger chunks (Windows Socket I/O is slow)
        # and continue by writing a single byte at a time so we can be sure we
        # completely fill the buffer.  Even though the socket API is allowed to
        # signal a short write via its return value it seems this doesn't
        # always happen on all platforms (FreeBSD and OS X particular) for the
        # very last bit of available buffer space.
        for msg in [b"x" * 65536, b"x"]:
            for i in range(1024 * 1024 * 64):
                try:
                    client_socket.send(msg)
                except OSError as e:
                    if e.errno == EWOULDBLOCK:
                        break
                    raise  # pragma: no cover
            else:  # pragma: no cover
                pytest.fail(
                    "Failed to fill socket buffer, cannot test BIO want write"
                )

        ctx = Context(SSLv23_METHOD)
        conn = Connection(ctx, client_socket)
        # Client's speak first, so make it an SSL client
        conn.set_connect_state()
        with pytest.raises(WantWriteError):
>           conn.do_handshake()

tests/test_ssl.py:3096:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
src/OpenSSL/SSL.py:2452: in do_handshake
    self._raise_ssl_error(self._ssl, result)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <OpenSSL.SSL.Connection object at 0x2b67e226250>, ssl = <cdata 'SSL *' 0x153011000>, result = -1

    def _raise_ssl_error(self, ssl: Any, result: int) -> None:
        if self._context._verify_helper is not None:
            self._context._verify_helper.raise_if_problem()
        if self._context._alpn_select_helper is not None:
            self._context._alpn_select_helper.raise_if_problem()
        if self._context._ocsp_helper is not None:
            self._context._ocsp_helper.raise_if_problem()

        error = _lib.SSL_get_error(ssl, result)
        if error == _lib.SSL_ERROR_WANT_READ:
>           raise WantReadError()
E           OpenSSL.SSL.WantReadError

src/OpenSSL/SSL.py:2037: WantReadError
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -40,8 +40,6 @@ jobs:
           - {VERSION: "3.11", NOXSESSION: "tests-cryptography-minimum"}
           - {VERSION: "3.12", NOXSESSION: "tests-cryptography-minimum"}
           - {VERSION: "3.13", NOXSESSION: "tests-cryptography-minimum"}
-          - {VERSION: "3.14-dev", NOXSESSION: "tests-cryptography-minimum"}
-          - {VERSION: "3.14t-dev", NOXSESSION: "tests-cryptography-minimum"}
           - {VERSION: "pypy-3.11", NOXSESSION: "tests-cryptography-minimum"}
           # Cryptography wheels
           - {VERSION: "3.9", NOXSESSION: "tests-cryptography-minimum-wheel"}

